ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -595,6 +595,10 @@ public static long getTtlFromRequestHeader(Map<String, Object> args) throws Rest
         throw new RestServiceException(Headers.TTL + "[" + ttlFromHeader + "] is not valid (has to be >= -1)",
             RestServiceErrorCode.InvalidArgs);
       }
+      if (ttlFromHeader == 0) {
+        throw new RestServiceException(Headers.TTL + "[" + ttlFromHeader + "] should not be 0",
+            RestServiceErrorCode.InvalidArgs);
+      }
```

Comment:
we probably don't need this one here.

Reply:
Remove this part and updated to add test and filter out null(default) after serialization as we discussed.